exercises/concept/karls-languages/.meta/design.md

```diff
@@ -43,6 +43,19 @@ This Concepts Exercise's Concepts are:
 
 This Concept Exercise's prerequisites Concepts are:
 
-- `for-loops`: know how to use a for-loop to iterate over a collection.
 - `arrays`: know of the array collection type and that it has a fixed length.
 - `strings`: data types used in this exercise
+
+## Analyzer
+
+This exercise could benefit from the following rules in the [analyzer]:
+
+- `essential`: If the student did not reuse the implementation of the `containsLanguage` method in the `isExciting` method, instruct them to do so.
+  Explain that reusing existing code instead of copy-pasting can help make code easier to maintain.
```

Comment:
I'm not sure that this is essential for this exercise. Although it's good practice to reuse code as much as possible, that's not the goal of the exercise. I'd change this to `actionable`.

Reply:
Actually I see that most community solutions use this:

```java
class LanguageList {
    boolean isExciting() {
        return languages.contains("Java") || languages.contains("Kotlin");
    }
}
```

In this case I would even go as far as saying that reusing `containsLanguage` instead of `language.contains()` becomes subjective enough that the comment might be `informative`...

Reply:
Percect, I agree with this change!